### PR TITLE
Only install requirements from PyPI when explicitly set

### PR DESCRIPTION
**Pull request by jhesketh**
Requirements should come from packages unless in developer mode or the user has explicitly set SOCOK8S_USE_VIRTUALENV.

### CODE REVIEW

File: script_library/pre-flight-checks.sh [hunk 1]
```diff
@@ -36,18 +36,38 @@ check_openstack_environment_is_ready_for_deploy (){
...
+
 check_ansible_requirements (){
+    if [[ "${SOCOK8S_USE_VIRTUALENV:-False}" == "True" ]]; then
+        install_ansible
+    fi
     # Ansible is required
-    which ansible-playbook > /dev/null || install_ansible
+    if ! which ansible-playbook > /dev/null 2>&1; then
+        echo "Ansible is not installed"
+        echo "OR set SOCOK8S_USE_VIRTUALENV=True to install ansible and other requirements"
```
Comment (jsuchome) [inline script_library/pre-flight-checks.sh]: Does not this miss first part of the sentence? ("Ansible is not installed" does not look like something that should be followed by OR)

Comment (jhesketh) [reply script_library/pre-flight-checks.sh]: Fixed.

**Comment by evrardjp** [comment]
Not tested but the code looks okay.

**Comment by jsuchome** [comment]
Replaced by https://github.com/SUSE-Cloud/socok8s/pull/378

**Pull request closed without merging**